### PR TITLE
feat(onboarding-tasks): Add endpoint to get onboarding tasks

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Problem**
Users completing tasks in the Quick Start may experience a strange situation where, after completing a task (e.g., sending a release), they still see the "Track releases" task as incomplete. This happens because there is no polling mechanism for the organization's data, where we retrieve the `onboardingTasks` property with the updated task statuses. The UI only reflects changes after a page refresh, which re-fetches the organization endpoint.

**Solution**
Since polling the organization endpoint is heavy (so I have heard many times), I am adding a dedicated `get` function to fetch only the onboarding tasks. This will allow us to safely poll for updates in the Quick Start sidebar, ensuring tasks are marked as complete without needing a full page refresh or re-fetch of the entire organization data.

Contributes to https://github.com/getsentry/projects/issues/248

### CODE REVIEW

File: src/sentry/api/endpoints/organization_onboarding_tasks.py [hunk 1]
```diff
@@ -64,3 +66,12 @@ def post(self, request: Request, organization) -> Response:
             onboarding_tasks.try_mark_onboarding_complete(organization.id)
 
         return Response(status=204)
+
+    def get(self, request: Request, organization) -> Response:
+
+        tasks_to_serialize = list(
+            onboarding_tasks.fetch_onboarding_tasks(organization, request.user)
+        )
```
Comment (evanpurkhiser) [inline src/sentry/api/endpoints/organization_onboarding_tasks.py:74]: Does this actually need to be wrapped with `list`? Is it a generator, can that just be passed directly into the serializer?

**Comment by evanpurkhiser** [comment]
How often are we planning to poll for these tasks?

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/api/endpoints/organization_onboarding_tasks.py (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/api/endpoints/organization_onboarding_tasks.py:74]: I copied it from here https://github.com/getsentry/sentry/blob/fc95869924926d555bdf3208a786fded56a28821/src/sentry/api/serializers/models/organization.py#L376-L377 so probably yes

Comment (evanpurkhiser) [reply src/sentry/api/endpoints/organization_onboarding_tasks.py:74]: probably fine

**Comment by priscilawebdev** [comment]
> How often are we planning to poll for these tasks?

I thought like we do for the project https://github.com/getsentry/sentry/blob/fc95869924926d555bdf3208a786fded56a28821/static/app/components/onboarding/useRecentCreatedProject.tsx#L10

File: tests/sentry/api/endpoints/test_organization_onboarding_tasks.py [hunk 2]
```diff
@@ -95,3 +95,22 @@ def test_missing_status_or_completion_seen(self):
...
             organization=self.org, task=OnboardingTask.FIRST_PROJECT
         ).exists()
+
+    def test_get_tasks(self):
+        # create a task
+        self.client.post(self.path, {"task": "create_project", "status": "complete"})
+
+        # get tasks
+        response = self.client.get(self.path)
+        assert response.status_code == 200, response.content
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply tests/sentry/api/endpoints/test_organization_onboarding_tasks.py:105]: for some reason if I use `self.get_success_response`, the post does not work anymore. Lets leave it as it is for now but it was good to know. thanks

File: src/sentry/api/endpoints/organization_onboarding_tasks.py (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/api/endpoints/organization_onboarding_tasks.py:73]: I get an error if I remove the list

**Pull request merged**